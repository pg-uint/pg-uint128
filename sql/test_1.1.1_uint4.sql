-- Testing cross-types compatibility

SELECT (0::uint4)::json;
SELECT json_build_object('some', 0::uint4);
SELECT ('0'::json)::uint4;
SELECT (0::uint4)::jsonb;
SELECT jsonb_build_object('some', 0::uint4);
SELECT ('0'::jsonb)::uint4;
SELECT ('"abc"'::jsonb)::uint4;
