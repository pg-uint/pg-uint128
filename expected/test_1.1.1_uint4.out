-- Testing cross-types compatibility
SELECT (0::uint4)::numeric;
 numeric 
---------
       0
(1 row)

SELECT ('0'::numeric)::uint4;
 uint4 
-------
 0
(1 row)

SELECT ('4294967296'::numeric)::uint4;
ERROR:  uint4 out of range
SELECT (0::uint4)::json;
 json 
------
 0
(1 row)

SELECT json_build_object('some', 0::uint4);
 json_build_object 
-------------------
 {"some" : 0}
(1 row)

SELECT ('0'::json)::uint4;
 uint4 
-------
 0
(1 row)

SELECT (0::uint4)::jsonb;
 jsonb 
-------
 0
(1 row)

SELECT jsonb_build_object('some', 0::uint4);
 jsonb_build_object 
--------------------
 {"some": 0}
(1 row)

SELECT ('0'::jsonb)::uint4;
 uint4 
-------
 0
(1 row)

SELECT ('"abc"'::jsonb)::uint4;
ERROR:  cannot cast jsonb string to type uint4
